pkg/version/version.go

```diff
@@ -67,3 +67,11 @@ func unknownString(s string) string {
 
 	return s
 }
+
+func unknownVersionString(s string) string {
+	if s == "" {
+		return "v.29.3"
```

Comment:
This would need to be updated continuously with each new release, and I don't think that's feasible.

Looking at one of the links you provided, and as @srenatus previously mentioned, this method looks more appealing https://github.com/vektra/mockery/blob/master/pkg/logging/logging.go#L36-L45

Would you want to give that a shot?

Reply:
I tried, but I wonder whether it will work: https://github.com/golang/go/issues/50603#issuecomment-2300983645

Locally I have tried:

```
func buildVersion() string {
	version, ok := debug.ReadBuildInfo()
	fmt.Println("----------", version.Main.Version)
	if ok && version.Main.Version != "(devel)" && version.Main.Version != "" {
		return version.Main.Version
	}
	return "unknown"
}
```

```
func New() Info {
	return Info{
		Version:   buildVersion(),
		GoVersion: goVersion,
		Platform:  platform,
		Commit:    unknownString(Commit),
		Timestamp: unknownString(Timestamp),
		Hostname:  unknownString(Hostname),
	}
}
```

and 

go build -ldflags "-s -w -X github.com/styrainc/regal/pkg/version.Version=1.2.3"

./regal version

```
---------- (devel)
Version:    unknown
Go Version: go1.23.3
Platform:   darwin/arm64
Commit:     unknown
Timestamp:  unknown
Hostname:   unknown
```